### PR TITLE
fix: remove Axios in favor of request

### PR DESCRIPTION
**Pull request by mattmeyerink**
MR to reapply [this reverted change](https://github.com/smartsheet/smartsheet-javascript-sdk/pull/36) that removes `request` in favor of `axios`. This MR fixes the issue where `PUT`/`POST` request configuration was being improperly passed along to `axios` causing authorization failures.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8310 +1,114 @@
 {
   "name": "smartsheet",
-  "version": "3.1.4",
-  "lockfileVersion": 2,
+  "version": "3.1.5",
+  "lockfileVersion": 1,
```
Comment (joshualoehr-smar) [inline package-lock.json]: This PR is modifying the lockfile version currently used, which is _probably_ fine but would perhaps be safer done in a separate PR. If you re-run npm install using Node 14.17.3 it will preserve the existing lockfile format.

Perhaps we could consider adding the volta pin to package.json from my closed PR a little while ago?

```json
  "volta": {
    "node": "14.17.3"
  }
```

File: lib/utils/httpRequestor.js [hunk 2]
```diff
@@ -99,59 +95,71 @@ exports.create = function(requestorConfig) {
...
+          callback(undefined, response.content);
+        }
+        
+        return response.content;
+      })
+      .catch((error) => {
+        logger.logErrorResponse(methodName, requestOptions, error);
+
+        if (callback) {
+          callback(error, undefined);
```
Comment (joshualoehr-smar) [inline lib/utils/httpRequestor.js:141]: I realize this is unchanged from your previous PR, but in looking at it again I'm wondering if the existing behavior is to strip off `headers` and `body` prior to passing the error to the callback? Is the `nodeify`'d callback passed the return value of `.catch` in Bluebird?

**Comment by joshualoehr-smar** [comment]
Has there been any test(s) added to cover the regression scenario we hit with the last PR?

File: lib/utils/httpRequestor.js [hunk 3]
```diff
@@ -99,59 +95,71 @@ exports.create = function(requestorConfig) {
...
-    makeRequest(method, methodName, requestOptions)
-      .catch(retryWithBackoffHelper(method, methodName, requestOptions, retryOptions, numRetries));
+  var retryHelper = (url, method, methodName, requestOptions, body, retryOptions, numRetries) => {
+    return methodHandler(url, method, methodName, requestOptions, body)
+      .then(handleResponse)
+      .catch(retryWithBackoffHelper(url, method, methodName, requestOptions, retryOptions, numRetries));
+  };
+  
+  var methodHandler = (url, method, methodName, requestOptions, body) => {
+    if (methodName === "POST" || methodName === "PUT") {
```
Comment (joshualoehr-smar) [inline lib/utils/httpRequestor.js:155]: Has there been any test(s) added to cover the regression scenario we hit with the last PR?

File: package-lock.json (hunk 1 above)
Comment (mattmeyerink) [reply package-lock.json]: The lockfile appears to have already been at version 2 so it must have been generated with a node version > 15. In that case to prevent us from going back to lockfile version 1 unintentionally should we volta pin to v16? What are your thoughts on this?

File: lib/utils/httpRequestor.js (hunk 3 above)
Comment (mattmeyerink) [reply lib/utils/httpRequestor.js:155]: Yes there is test coverage to ensure this logic is functional. Originally there was coverage to catch the contract with the http request library but the issue surfaced from incorrectly updating those tests 😅. When I added the fix for this issue there were 3 tests failing for `POST` and 3 tests failing for `PUT` that I updated to reflect the correct `axios` contract so we should be covered again moving forward 😄

File: package-lock.json (hunk 1 above)
Comment (joshualoehr-smar) [reply package-lock.json]: Looks like that was inadvertently done during the package bump for the revert (3.1.4) https://github.com/smartsheet/smartsheet-javascript-sdk/commit/8675c13a8711ee7d343386a2c6e738d125283e7b

Kind of a tough call in this case. Like I said, it's _probably_ fine to leave it as is? Theoretically npm should be able to modify its own lockfile schema without any breaking changes to the library code. I do think we should pin to _something_ so this doesn't happen again. 16 would be a good choice, just a matter of how much extra risk to add to this PR. I assume the current lockfile was made with 16.x, so maybe we just consider ourselves upgraded and stick with that.

File: lib/utils/httpRequestor.js (hunk 2 above)
Comment (mattmeyerink) [reply lib/utils/httpRequestor.js:141]: Looking at this again I believe that this change preserves the initial callback behavior. The two parameters accepted are an `error` and a `success` object. In the case of failure there isn't going to be a success response from the promise chain. If the network request was the cause of the failure the body/headers should be funneled into the `error` object passed into the callback which would be the behavior I would expect from `nodeify`.

File: package-lock.json (hunk 1 above)
Comment (mattmeyerink) [reply package-lock.json]: Yeah I agree the safest option is likely to leave it at version 2 and pin to node 16 for future development. This will limit the `package-lock.json` changes to removing `request` and adding `axios`. Making this change now.

**Review by joshualoehr-smar — APPROVED** [review]

**Review by Scomodev — APPROVED** [review]
